src/pages/studyView/StudyViewPage.tsx

```diff
@@ -47,16 +42,34 @@ export default class StudyViewPage extends React.Component<IStudyViewPageProps,
 
                 this.store.updateStoreFromURL(query);
             },
-            { fireImmediately: true }
+            {fireImmediately: true}
         );
     }
 
     private handleTabChange(id: string) {
-        this.props.routing.updateRoute({ tab: id });
+        this.props.routing.updateRoute({tab: id});
     }
 
-    content(){
+    @computed
+    get numberOfSamplesWithMutation() {
+        if (this.store.molecularProfileSampleCounts.isComplete &&
+            this.store.molecularProfileSampleCounts.result.numberOfMutationProfiledSamples) {
+            return this.store.molecularProfileSampleCounts.result.numberOfMutationProfiledSamples;
+        } else {
+            return 0;
+        }
+    }
 
+    @computed
```

Comment:
You should avoid referencing mobx promises inside computed functions.  The problem is that they hide load status.  Instead we should accept the results of the promises as arguments to a pure function and handle loading status outside.   For example, this function will return 0 inaccurately.  It's not 0, it just hasn't loaded yet.   I know this is a pain in the ass but it's just not a good practice.

Reply:
Makes sense, will update